Sources/SwiftLanguageService/ParametersDocumentationExtractor.swift

```diff
@@ -0,0 +1,192 @@
+import Foundation
+import Markdown
+
+private struct ParametersDocumentationExtractor {
+  private var parameters = [String: String]()
+
+  /// Extracts parameter documentation from a markdown string.
+  ///
+  /// - Returns: A tuple containing the extracted parameters and the remaining markdown.
+  mutating func extract(from markdown: String) -> (parameters: [String: String], remaining: String) {
+    let document = Document(parsing: markdown, options: [.parseBlockDirectives, .parseMinimalDoxygen])
+
+    var remainingBlocks = [any BlockMarkup]()
+
+    for block in document.blockChildren {
+      switch block {
+      case let unorderedList as UnorderedList:
+        if let newUnorderedList = extract(from: unorderedList) {
+          remainingBlocks.append(newUnorderedList)
+        }
+      case let doxygenParameter as DoxygenParameter:
+        extract(from: doxygenParameter)
+      default:
+        remainingBlocks.append(block)
+      }
+    }
+
+    let remaining = Document(remainingBlocks).format()
+
+    return (parameters, remaining)
+  }
+
+  /// Extracts parameter documentation from a Doxygen parameter command.
+  private mutating func extract(from doxygenParameter: DoxygenParameter) {
+    parameters[doxygenParameter.name] = Document(doxygenParameter.blockChildren).format()
+  }
+
+  /// Extracts parameter documentation from an unordered list.
+  ///
+  /// - Returns: A new UnorderedList with the items that were not added to the parameters if any.
+  private mutating func extract(from unorderedList: UnorderedList) -> UnorderedList? {
+    var newItems = [ListItem]()
+
+    for item in unorderedList.listItems {
+      if extractSingle(from: item) || extractOutline(from: item) {
+        continue
+      }
+
+      newItems.append(item)
+    }
+
+    if newItems.isEmpty {
+      return nil
+    }
+
+    return UnorderedList(newItems)
+  }
+
+  /// Parameter documentation from a `Parameters:` outline.
+  ///
+  /// Example:
+  /// ```markdown
+  /// - Parameters:
+  ///   - param: description
+  /// ```
+  ///
+  /// - Returns: True if the list item has parameter outline documentation, false otherwise.
+  private mutating func extractOutline(from listItem: ListItem) -> Bool {
+    guard let firstChild = listItem.child(at: 0) as? Paragraph,
```

Comment:
Just checking: How does `child(at:)` behave if there is no element in the list? (Not sure why we would have parsed a list in that case but still). If it returns `nil`, then fine, otherwise, it would be nice to check the number of children in `listItem` and `firstChild`.

Reply:
It does return `nil` when the index passed is ≥ the children count.